### PR TITLE
Project specific and target specific library type override

### PR DESCRIPTION
**Pull request by awnawab**
This PR adds the capability to override library types across entire projects or for individual targets using the following nomenclature:

```
<PROJECT-NAME-UPPER-CASE>_BUILD_SHARED_LIBS=OFF/ON
ECBUILD_TARGET_<target_name>_TYPE=STATIC/SHARED/OBJECT/MODULE/INTERFACE
```

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf/ecbuild?pullRequest=60) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf/ecbuild?pullRequest=60) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf/ecbuild?pullRequest=60) it.</sub>

File: cmake/ecbuild_project.cmake [hunk 1]
```diff
@@ -71,6 +71,12 @@ macro( project _project_name )
 
     ecbuild_declare_project()
 
+    ### override BUILD_SHARED_LIBS property
+    if( DEFINED ${PNAME}_SHARED_LIBS )
+       set( BUILD_SHARED_LIBS ${${PNAME}_SHARED_LIBS} )
```
Comment (wdeconinck) [inline cmake/ecbuild_project.cmake:76]:
```suggestion
    if( DEFINED ${PNAME}_BUILD_SHARED_LIBS )
       set( BUILD_SHARED_LIBS ${${PNAME}_BUILD_SHARED_LIBS} )
```

**Comment by wdeconinck** [comment]
Hi @awnawab that's a good improvement. I don't see immediately the test with
```-DBUILD_SHARED_LIBS=ON -D<PNAME>_BUILD_SHARED_LIBS=ON```

**Comment by awnawab** [comment]
Hi @wdeconinck. Very well spotted, that is definitely a test we need. I'll add it, thanks!

**Pull request merged**